tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+					p.MountPath, "usr", "lib64")).CombinedOutput()
+			_, _ = exec.Command(
+				"cp", "/lib64/libnss_files.so.2", filepath.Join(
+					p.MountPath, "usr", "lib64")).CombinedOutput()
+
+			_, _ = exec.Command(
+				"cp", "bin/amd64/tests/id", filepath.Join(
+					p.MountPath, "usr", "bin")).CombinedOutput()
+			_, _ = exec.Command(
+				"cp", "bin/amd64/tests/useradd", filepath.Join(
+					p.MountPath, "usr", "sbin")).CombinedOutput()
+			_, _ = exec.Command(
+				"cp", "bin/amd64/tests/usermod", filepath.Join(
+					p.MountPath, "usr", "sbin")).CombinedOutput()
+		}
+	}
+}
+
+func getRootLocation(t *testing.T, partitions []*Partition) string {
+	for _, p := range partitions {
+		if p.Label == "ROOT" {
+			return p.MountPath
+		}
+	}
+	t.Fatal("ROOT filesystem not found! A partition labeled ROOT is requred")
+	return ""
+}
+
+func removeFile(t *testing.T, imgName string) {
+	err := os.Remove(imgName)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func removeMountFolders(t *testing.T, partitions []*Partition) {
+	for _, p := range partitions {
+		err := os.RemoveAll(p.MountPath)
+		if err != nil {
+			t.Log(err)
+		}
+	}
+}
+
+func runIgnition(t *testing.T, stage string, root string) {
+	out, err := exec.Command(
+		"ignition", "-clear-cache", "-oem",
+		"file", "-stage", stage, "-root", root).CombinedOutput()
+	debugInfo, derr := ioutil.ReadFile("/var/log/syslog")
+	if derr == nil {
+		debugOut := []string{}
+		lines := strings.Split(string(debugInfo), "\n")
+		for _, line := range lines {
+			if strings.Contains(line, "ignition") {
+				debugOut = append(debugOut, line)
+			}
+		}
+	}
+	if err != nil {
+		t.Fatal("ignition", err, string(out))
+	}
+
+}
+
+func pickDevice(t *testing.T, partitions []*Partition,
+	fileName string, label string) string {
+	number := -1
+	for _, p := range partitions {
+		if p.Label == label {
+			number = p.Number
+		}
+	}
+	if number == -1 {
+		t.Fatal("Didn't find a drive with label:", label)
+		return ""
+	}
+
+	kpartxOut, err := exec.Command("kpartx", "-l", fileName).CombinedOutput()
+	if err != nil {
+		t.Fatal("kpartx -l", err, string(kpartxOut))
+	}
+	return fmt.Sprintf("/dev/mapper/%sp%d",
+		strings.Trim(strings.Split(string(kpartxOut), " ")[4], "/dev/"), number)
+}
+
+func writeIgnitionConfig(t *testing.T, config string) {
+	err := ioutil.WriteFile("config.ign", []byte(config), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func calculateImageSize(partitions []*Partition) int64 {
+	size := int64(63 * 512)
+	for _, p := range partitions {
+		size += int64(align(p.Length, 512) * 512)
+	}
+	size = size + int64(4096*512) // extra room to allow for alignments
+	return size
+}
+
+func createVolume(
+	t *testing.T, fileName string, size int64, cylinders int, heads int,
+	sectorsPerTrack int, partitions []*Partition) {
+	// attempt to create the file, will leave already existing files alone.
+	// os.Truncate requires the file to already exist
+	out, err := os.Create(fileName)
```

Comment:
Oh wait, I now see that you're truncating it to a non-zero size, and after reading the docs for `os.Create` it says:

> Create creates the named file with mode 0666 (before umask), truncating it if it already exists

I think you'll want to call `OpenFile` with just `os.O_RDWR` and `os.O_CREATE` to prevent it from truncating it to 0, unless that isn't an issue here.